### PR TITLE
Handle change events

### PR DESCRIPTION
**Pull request by gsvr**
- [x] Change message format & times

### CODE REVIEW

**Comment by gsvr** [comment]
@imsickofmaps Please review.

File: changes/tasks.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+#     headers = {'Authorization': ['Token %s' % settings.IDENTITIES_TOKEN],
+#                'Content-Type': ['application/json']}
+#     r = requests.get(url, headers=headers)
+#     return r.json()
+
+
+def deactivate_subscription(subscription):
+    """ Sets a subscription deactive via a Patch request
+    """
+    url = settings.STAGE_BASED_URL + 'subscriptions/%s/' % subscription["id"]
```
Comment (imsickofmaps) [inline changes/tasks.py]: Please make prefix `STAGE_BASED_MESSAGING_` for consistency

File: changes/tasks.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+def get_messageset(messageset_id):
+    url = settings.STAGE_BASED_URL + 'messageset/%s/' % messageset_id
+    headers = {'Authorization': ['Token %s' % settings.STAGE_BASED_TOKEN],
+               'Content-Type': ['application/json']}
+    r = requests.get(url, headers=headers)
+    return r.json()
+
+
+# def unsubscribe_identity(identity, reason):
```
Comment (imsickofmaps) [inline changes/tasks.py]: Remove?

File: changes/tasks.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+def get_subscription(identity):
+    """ Gets the first active subscription found for an identity
+    """
+    url = settings.STAGE_BASED_URL + 'subscriptions/'
+    params = {'id': identity, 'active': True}
+    headers = {'Authorization': ['Token %s' % settings.STAGE_BASED_TOKEN],
+               'Content-Type': ['application/json']}
+    r = requests.get(url, params=params, headers=headers)
+    return r.json()[0]  # return first object TODO: handle multiple
```
Comment (imsickofmaps) [inline changes/tasks.py]: This response will be: `
{
    "count": 0,
    "next": null,
    "previous": null,
    "results": []
}`

File: changes/tasks.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+    def change_baby(self, change):
+        pass
+
+    def change_loss(self, change):
+        pass
+
+    def change_messaging(self, change):
+        # Get mother's current subscription
+        subscription = get_subscription(change.mother_id)
```
Comment (imsickofmaps) [inline changes/tasks.py]: Make `subscriptions`

File: changes/tasks.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+        # Deactivate subscription
+        subscription = deactivate_subscription(subscription)
```
Comment (imsickofmaps) [inline changes/tasks.py]: for loop or open an issue to track the to-do please

File: changes/tasks.py [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+        # if the registration was for postbirth, we can assume postbirth
+        if registration.stage == 'postbirth':
+            stage = 'postbirth'
+            weeks = calc_baby_age(get_today(), registration.data["baby_dob"])
+        # otherwise, we need to look if the user has changed to baby
+        else:
+            baby_switch = Change.objects.filter(mother_id=change.mother_id,
+                                                action='change_baby')
+            if baby_switch.count() > 0:
+                # TODO: handle a person that has switched to baby for a
```
Comment (imsickofmaps) [inline changes/tasks.py]: Open issue please for "General multi-preg handling" with a link to this

File: changes/tasks.py [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+
+    def unsubscribe_mother_only(self, change):
+        pass
+
+    def run(self, change_id, **kwargs):
+        """ Implements the appropriate action
+        """
+        change = Change.objects.get(id=change_id)
+
+        if change.action == 'change_baby':
```
Comment (imsickofmaps) [inline changes/tasks.py]: for giant elif's like this I prefer: 

```
result = {
    'change_baby': self.change_baby,
    'change_loss': self.change_loss,
    'change_messaging': self.change_messaging,
    'change_language': self.change_language,
    'unsubscribe_household_only': self.unsubscribe_household_only,
    'unsubscribe_mother_only': self.unsubscribe_mother_only,
}.get(change.action, None)(change)

```

File: hellomama_registration/settings.py [hunk 8]
```diff
@@ -138,6 +139,7 @@
 }
 
 HOOK_DELIVERER = 'registrations.tasks.deliver_hook_wrapper'
+# TODO: Add changes hook deliverer?
```
Comment (imsickofmaps) [inline hellomama_registration/settings.py]: shared. Can remove.

File: hellomama_registration/settings.py [hunk 9]
```diff
@@ -187,4 +193,8 @@
                                  'http://localhost:8005/api/v1/')
 STAGE_BASED_TOKEN = os.environ.get('STAGE_BASED_TOKEN', 'REPLACEME')
 
+IDENTITIES_URL = os.environ.get('IDENTITIES_URL',
```
Comment (imsickofmaps) [inline hellomama_registration/settings.py]: `IDENTITY_STORE_` prefix for consistency please

File: registrations/tasks.py [hunk 10]
```diff
@@ -130,7 +130,18 @@ def get_messageset_schedule_sequence(short_name, weeks, voice_days,
...
-    next_sequence_number = msgs_per_week * weeks
+    # check if in prebirth stage - only starting messaging in week 10
+    if 'prebirth' in short_name:
+        next_sequence_number = msgs_per_week * (
+            weeks - settings.PREBIRTH_MIN_WEEKS)
+        if next_sequence_number <= 0:
+            next_sequence_number = 1
+    elif '13_52' in short_name:
+        next_sequence_number = msgs_per_week * (weeks - 13)
+        if next_sequence_number <= 0:
```
Comment (imsickofmaps) [inline registrations/tasks.py]: make this `== 0` please. Rounding issues are fine, but if it goes to minus values something else has gone wrong and we want an exception in place.

**Comment by imsickofmaps** [comment]
First pass done

**Comment by imsickofmaps** [comment]
:+1:

**Pull request merged**